sympy/functions/elementary/exponential.py

```diff
@@ -402,7 +402,10 @@ def _eval_is_extended_real(self):
             return arg2.is_even
 
     def _eval_is_complex(self):
-        return self.args[0].is_complex
+        def complex_extended_negative(arg):
+            yield arg.is_complex
+            yield arg.is_extended_negative
+        return fuzzy_or(complex_extended_negative(self.args[0]))
```

Comment:
How about writing

    arg = self.args[0]
    return fuzzy_or(arg.is_complex, arg.is_extended_negative)

Reply:
I could do. I wrote it like this because we can delay evaluation of is_extended_negative. When `is_complex` gives true `is_extended_negative` is not evaluated. I would like to rewrite lots of the core assumption routines in this way.

This example is particularly simple and probably doesn't matter too much but some of the more complicated routines like `Pow._eval_is_*` could probably be simplified and made more correct using this style.

Reply:
> When is_complex gives true is_extended_negative is not evaluated.

+1